samples/cellular/modem_shell/src/sock/sock.c

```diff
@@ -547,11 +558,20 @@ int sock_open_and_connect(
 	}
 
 	/* Get address */
-	err = sock_getaddrinfo(socket_info, family, type, address, port, pdn_cid);
+	err = sock_getaddrinfo_req(&socket_info->addrinfo, family, type, address, pdn_cid);
 	if (err) {
 		goto connect_error;
 	}
 
+	/* Set port to address info */
+	if (family == AF_INET) {
+		((struct sockaddr_in *)socket_info->addrinfo->ai_addr)->sin_port = htons(port);
+	} else if (family == AF_INET6) {
+		((struct sockaddr_in6 *)socket_info->addrinfo->ai_addr)->sin6_port = htons(port);
+	} else {
+		assert(0);
```

Comment:
@stig-bjorlykke I need to remove this assert because raw sockets are hitting it always.

Reply:
@trantanen I did not add this assert, I only moved the code. It's ok for me.

Reply:
Yep, moving the code doesn't work here for raw sockets that were treated separately in the previous location.
And while reviewing the code, Github collapses some important conditions from the previous location so it was tough to figure this out.
Fixing in PR #15723.